reference/commands/lock/add.rst

```diff
@@ -29,3 +29,89 @@ conan lock add
       --lockfile-out LOCKFILE_OUT
                             Filename of the created lockfile
       --lockfile LOCKFILE   Filename of the input lockfile
+
+The ``conan lock add`` command is able to add a package version to an existing or new lockfile ``requires``, ``build_requires`` or ``python_requires``.
+
+For example, the following is able to create the following lockfile (by default, named ``conan.lock``):
+
+.. code-block:: bash
+
+  $ conan lock add --requires=pkg/1.1 --build-requires=tool/2.2 --python-requires=mypytool/3.3 
+  Generated lockfile: ...conan.lock
+
+  $cat conan.lock
+  {
+      "version": "0.5",
+      "requires": [
+          "pkg/1.1"
+      ],
+      "build_requires": [
+          "tool/2.2"
+      ],
+      "python_requires": [
+          "mypytool/3.3"
+      ]
+  }
+
+
+If an input profile is provided, the command will add the arguments to the existing lockfile, and maintain the
+package versions sorted:
+
+.. code-block:: bash
+
+  $ conan lock add --build-requires=tool/2.3 --lockfile=conan.lock
+  Using lockfile: '.../conan.lock'
+  Generated lockfile: .../conan.lock
+
+  $ cat conan.lock
+  {
+      "version": "0.5",
+      "requires": [
+          "pkg/1.1"
+      ],
+      "build_requires": [
+          "tool/2.3",
+          "tool/2.2"
+      ],
+      "python_requires": [
+          "mypytool/3.3"
+      ]
+  }
+
+
+The ``conan lock add`` command does not perform any checking on the lockfile, the packages, the existence of packages,
+the existence of package versions, or the existence of those packages in a given dependency graph, it is a basic manipulation of the json information.
+When that lockfile is applied to resolve a dependency graph, it is possible that the added versions do not exist,
+or do not resolve for the ``conanfile.py`` recipes defined version ranges.
+
+Moreover, the list of versions is still sorted. Adding an older version like ``tool/2.1`` won't make that version being used
+automatically if the recipes contain the version range ``tool/[>=2.0 <3]``, if the ``tool/2.2`` is listed there.
+
+Note that a lockfile created with ``conan lock add`` might be incomplete if it doesn't define all possible versions in a dependency graph.
+For those cases, recall that the ``--lockfile-partial`` argument can be applied. Note also that if a ``conan.lock`` file exist in the
```

Comment:
I think we should link this here: https://docs.conan.io/2/tutorial/versioning/lockfiles.html#multi-configuration-lockfiles or extend the explanation because it sounds a bit confusing.